CDPBatchEditor/Commands/Command/OverrideCommand.cs

```diff
@@ -0,0 +1,127 @@
+
+namespace CDPBatchEditor.Commands.Command
+{
+    using System;
+    using System.Linq;
+
+    using CDP4Common.EngineeringModelData;
+
+    using CDP4Dal.Operations;
+
+    using CDPBatchEditor.CommandArguments.Interface;
+    using CDPBatchEditor.Commands.Command.Interface;
+    using CDPBatchEditor.Services.Interfaces;
+
+    /// <summary>
+    /// Defines an <see cref="OverrideCommand" /> that provides actions that are
+    /// <see cref="CDP4Common.EngineeringModelData.ParameterOverride" /> related
+    /// </summary>
+    public class OverrideCommand : IOverrideCommand
+    {
+        /// <summary>
+        /// Gets the injected <see cref="ICommandArguments" /> instance
+        /// </summary>
+        private readonly ICommandArguments commandArguments;
+
+        /// <summary>
+        /// Gets the injected <see cref="IFilterService" /> instance
+        /// </summary>
+        private readonly IFilterService filterService;
+
+        /// <summary>
+        /// Gets the injected <see cref="ISessionService" /> instance
+        /// </summary>
+        private readonly ISessionService sessionService;
+
+        /// <summary>
+        /// Initialise a new <see cref="OverrideCommand" />
+        /// </summary>
+        /// <param name="commandArguments">the <see cref="ICommandArguments" /> arguments instance</param>
+        /// <param name="sessionService">
+        /// the <see cref="ISessionService" /> providing the <see cref="CDP4Dal.ISession" /> for the
+        /// application
+        /// </param>
+        /// <param name="filterService">the <see cref="IFilterService" /></param>
+        public OverrideCommand(ICommandArguments commandArguments, ISessionService sessionService, IFilterService filterService)
+        {
+            this.commandArguments = commandArguments;
+            this.sessionService = sessionService;
+            this.filterService = filterService;
+        }
+
+        /// <summary>
+        /// Override parameters with given short names, possibly filtered on Ownership (Included Owner) or Category.
+        /// </summary>
+        public void Override()
+        {
+            if (!this.commandArguments.SelectedParameters.Any())
+            {
+                Console.WriteLine("No --parameters given. Override parameters skipped.");
+                return;
+            }
+
+            var allElementUsages = this.sessionService.Iteration.Element.SelectMany(x => x.ContainedElement).ToArray();
+
+            foreach (var elementDefinition in this.sessionService.Iteration.Element
+                .Where(e => this.filterService.IsFilteredIn(e))
+                .OrderBy(x => x.ShortName).ToArray())
+            {
+                // From 10-25 docs:
+                // Note 2: The owner DomainOfExpertise of this ParameterOverride is the same as the owner of the elementDefinition. 
+                var overrider = elementDefinition.Owner;
```

Comment:
this can also be another domain, i think it should be the owner of the Parameter this being overriden. That is the use case we have now